src/include/gurt/telemetry_consumer.h

```diff
@@ -49,9 +49,9 @@ int d_tm_list(struct d_tm_context *ctx, struct d_tm_nodeList_t **head,
 int d_tm_list_subdirs(struct d_tm_context *ctx, struct d_tm_nodeList_t **head,
 		      struct d_tm_node_t *node, uint64_t *node_count,
 		      int max_depth);
-void d_tm_print_my_children(struct d_tm_context *ctx, struct d_tm_node_t *node,
-			    int level, int filter, char *path, int format,
-			    int opt_fields, FILE *stream);
+void d_tm_iterate(struct d_tm_context *ctx, struct d_tm_node_t *node,
```

Comment:
Any reason not to keep the original `d_tm_print_my_children` as a wrapper, and add a `d_tm_reset_children`  wrapper as well? `d_tm_iterate` is a good internal implementation, but it seems a good idea to simplify the interface used by external callers.

Reply:
Hmm, I thought there might be cases to do multiple operations in one iteration, for example print & reset.  So if we use different wrapper for each operation,  it will do multiple iteration.  So I choose to export d_tm_iterate to utils.

Reply:
That's fair. Also based on your patch, it doesn't look like it's used in many places currently anyway.